pkgs/applications/terminal-emulators/mlterm/default.nix

```diff
@@ -62,20 +139,15 @@ stdenv.mkDerivation rec {
   '';
 
   configureFlags = [
-    "--with-imagelib=gdk-pixbuf" #or mlimgloader depending on your bugs of choice
-    "--with-type-engines=cairo,xft,xcore"
-    "--with-gtk=3.0"
-    "--enable-ind" #indic scripts
-    "--enable-fribidi" #bidi scripts
-    "--with-tools=mlclient,mlconfig,mlcc,mlterm-menu,mlimgloader,registobmp,mlfc"
-    #mlterm-menu and mlconfig depend on enabling gnome.at-spi2-core
-    #and configuring ~/.mlterm/key correctly.
-  ] ++ lib.optionals (!stdenv.isDarwin) [
-    "--with-x=yes"
-    "--with-gui=xlib,fb"
-    "--enable-m17nlib" #character encodings
-  ] ++ lib.optionals stdenv.isDarwin [
-    "--with-gui=quartz"
+    "--with-imagelib=${imagelibChoice}"
+    "--with-type-engines=${lib.concatStringsSep "," (builtins.attrNames (lib.filterAttrs (n: v: v) typeEnginesChoices))}"
```

Comment:
Nit: That's function took a bit to understand and you're using it three times. Extracting it into a var with a descriptive name would be nice.

Reply:
I agree, done.